tests/framework/statistics/consumer.py

```diff
@@ -92,27 +126,46 @@ def process(self, check_criteria=True) -> (dict, dict):
             st_defs = self._measurements_defs[ms_name].statistics
             for st_def in st_defs:
                 if st_def.name not in self._results[ms_name]:
-                    assert Consumer.DATA_KEY in self._results[ms_name], \
-                        "Results does not have extracted data points for" \
-                        f" {ms_name} measurement."
+                    if Consumer.DATA_KEY not in self._results[ms_name]:
+                        raise Exception("Results does not have extracted data "
+                                        f"points for {ms_name} measurement.")
                     self._statistics[ms_name][st_def.name] = \
-                        st_def.func(self._results[ms_name][self.DATA_KEY])
+                        self._statistics[ms_name][st_def.name] = {
+                            "value": st_def.func(self._results[ms_name][
+                                                 self.DATA_KEY])
+                        }
                 else:
-                    self._statistics[ms_name][st_def.name] = self._results[
-                        ms_name][st_def.name]
+                    self._statistics[ms_name][st_def.name] = {
+                        "value": self._results[ms_name][st_def.name]
+                    }
 
                 pass_criteria = st_def.pass_criteria
-                if pass_criteria and check_criteria:
-                    res = self._statistics[ms_name][st_def.name]
+                if pass_criteria:
+                    self._statistics[ms_name][st_def.name][
+                        "pass_criteria"] = {
+                            pass_criteria.name: pass_criteria.baseline
+                        }
+                    res = self._statistics[ms_name][st_def.name]["value"]
                     try:
                         pass_criteria.check(res)
-                    except Failed as err:
-                        self._reset()
+                        self._statistics[ms_name][st_def.name]["outcome"] = \
+                            "PASSED"
+                    except CriteriaException as err:
                         # pylint: disable=W0707
-                        raise Failed(msg=f"'{ms_name}/{st_def.name}':"
-                                         f" {err.msg}")
+                        self._statistics[ms_name][st_def.name]["outcome"] = \
+                            "FAILED"
+                        fail_msg = f"'{ms_name}/{st_def.name}': {err}"
+                        self._failure_aggregator.add_message(fail_msg)
+                        if fail_fast:
+                            raise self._failure_aggregator
```

Comment:
I think here we can directly raise the CriteriaException

Reply:
What I meant is that on this `fail_fast` branch (`if fail_fast:`), we can directly return the `CriteriaException`, since this is the only failure present in the `_failure_aggregator`. If there was another one, it would have already been raised by this same code. What I mean is that we don't ned a `FailureAggregator` for this `fail_fast` mechanism. It only makes sense when `fail_fast` is off

Reply:
I think this makes sense